server/src/main/java/org/apache/iotdb/db/engine/cache/TimeSeriesMetadataCache.java

```diff
@@ -134,7 +142,7 @@ public TimeseriesMetadata get(TimeSeriesMetadataCacheKey key, Set<String> allSen
       timeSeriesMetadataList.forEach(timeseriesMetadata ->
           lruCache.put(new TimeSeriesMetadataCacheKey(key.filePath, key.device,
               timeseriesMetadata.getMeasurementId()), timeseriesMetadata));
-      return lruCache.get(key);
+      return new TimeseriesMetadata(lruCache.get(key));
```

Comment:
As we already replace the reader by resource, do we need to copy? 
But the resource may be also removed from memory, so copy is fine...